File: UniquePath/Memoization.py
def UniquePath(m,n,dp):
    if m == 0 and n == 0: return 1
    if m < 0 or n < 0: return 0
    if dp[m][n] != -1: return dp[m][n]

    top = UniquePath(m-1,n,dp)
    left = UniquePath(m,n-1,dp)

    dp[m][n] = top + left
    return dp[m][n]

if __name__ == "__main__":
    m = 3 ; n =7
    dp = [ [ -1 for _  in range(n) ] for _ in range(m)]
    
    print(UniquePath(m-1,n-1,dp))

File: UniquePath/Memoization.java

import java.util.Arrays;

class Memoization {

    public static int findUniquePath(int m,int n,int[][] dp) {
        if(m == 0 && n == 0) return 1;
        if(m < 0 || n < 0) return 0;
        if(dp[m][n] != -1) return dp[m][n];

        int top =  findUniquePath(m-1, n,dp);
        int left = findUniquePath(m,n-1,dp);

        return dp[m][n]  = left + top;
    }

    public static void main(String[] args) {
        int m = 3, n = 7;
        int[][] dp = new int[m][n];

        for(int i = 0 ; i < m ;i++) {
            Arrays.fill(dp[i],-1);
        }

        System.out.println(findUniquePath(m-1,n-1,dp));
    }
}